copado-function/app/Deploy.js

```diff
@@ -258,14 +258,24 @@ async function run() {
     // and commit it to the repo as a backup
     const gitDiffArr = await Deploy.retrieveAndCommit(targetBU, commitSelectionArr);
 
-    try {
-        Log.info('git-push changes');
-        Log.info('===================');
-        Util.push(CONFIG.mainBranch);
-    } catch (ex) {
-        Log.info('git push failed: ' + ex.message);
-        throw ex;
-    }
+    // trying to push
+    let success = false,
+        i = 0;
+    do {
+        try {
+            Log.info('git-push changes');
+            Log.info('===================');
+            Util.push(CONFIG.mainBranch);
+            success = true;
+        } catch (ex) {
+            if (ex.message === 'Error: Command failed: git push origin "master"') {
+                Util.execCommand(null, ['git fetch origin "' + CONFIG.mainBranch + '"'], null);
+                Util.execCommand(null, ['git reset --hard origin/' + CONFIG.mainBranch], null);
+                Util.execCommand(null, ['git merge "' + CONFIG.promotionBranch + '"'], null);
+            }
+        }
+        i++;
```

Comment:
this should be moved to the top of your catch block because if anything fails inside of the catch this will never be reached and we have an endless loop